### ISSUE TITLE
Server returns inconsistent results when S3 credentials are invalidated

### ISSUE THREAD

**Issue by rmliddle** [quickwit-oss/quickwit#332]
**Describe the bug**

When S3 credentials are invalidated, the server appears to return a partial result where numHits is non-zero but hits are not present
```bash
> GET /api/v1/hdfs_logs/search?query=severity_text:ERROR&startTimestamp=1442834249&endTimestamp=1442900000 HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.54.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< content-type: application/json
< content-length: 59
< date: Sat, 24 Jul 2021 01:35:09 GMT
< 
{
  "numHits": 6,
  "hits": [],
  "numMicrosecs": 3802390
* Connection #0 to host localhost left intact
}
```
Subsequent requests appear to then return an error message (along with a 500 status code).  

**Steps to reproduce (if applicable)**
After indexing the queryset in this [tutorial](https://quickwit.io/docs/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3)
1. Issue the above request
2. Invalidate S3 credentials + reissue request

**Expected behavior**
There should presumably be either a correct result or error present in the response

**System configuration:**
macOS Mojave
version 10.14.6
```
> quickwit -V
Quickwit 0.1.0
```

**Comment by fulmicoton** [comment]
Thank you for the report @rmliddle !

**Comment by fmassot** [comment]
@mosuka can you test this on s3?

**Comment by mosuka** [comment]
@fmassot Yes, I’ll investigate it.

**Comment by fmassot** [comment]
I think the issue is not related specifically to S3 and is coming from this part of `root_search`:
```rust
    // Merge the fetched docs.
    let mut hits: Vec<Hit> = Vec::new();
    for response in fetch_docs_responses {
        match response {
            Ok(fetch_docs_result) => {
                hits.extend(fetch_docs_result.hits);
            }
            // TODO handle failure.
            Err(err) => error!(err=?err),
        }
    }
```

I will open an issue to handle property this part.

**Comment by fmassot** [comment]
see #380

**Comment by fmassot** [comment]
@rmliddle this is fixed on the main branch thanks to #530. I will leave the issue open until the next release.

**Comment by fmassot** [comment]
Closing as 0.2 has been released.
